Sources/Hummingbird/Server/Baggage.swift

```diff
@@ -0,0 +1,196 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Hummingbird server framework project
+//
+// Copyright (c) 2021-2023 the Hummingbird authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See hummingbird/CONTRIBUTORS.txt for the list of Hummingbird authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Tracing
+
+extension HBRequest {
+    /// Baggage attached to request. Used to propagate baggage to child functions
+    var baggage: Baggage {
+        get { self.extensions.get(\.baggage) ?? Baggage.topLevel }
+        set { self.extensions.set(\.baggage, value: newValue) }
+    }
+
+    /// Execute the given operation with edited request that includes baggage
+    ///
+    /// - Parameters:
+    ///   - baggage: Baggage to attach to request
+    ///   - operation: operation to run
+    /// - Returns: return value of operation
+    func withBaggage<Return>(_ baggage: Baggage, _ operation: (HBRequest) throws -> Return) rethrows -> Return {
+        var request = self
+        request.baggage = baggage
+        return try operation(request)
+    }
+
+    /// Execute the given operation within a newly created ``Span``,
+    ///
+    /// DO NOT `end()` the passed in span manually. It will be ended automatically when the `operation` returns.
+    ///
+    /// - Parameters:
+    ///   - operationName: The name of the operation being traced. This may be a handler function, database call, ...
+    ///   - kind: The ``SpanKind`` of the ``Span`` to be created. Defaults to ``SpanKind/internal``.
+    ///   - operation: operation to wrap in a span start/end and execute immediately
+    /// - Returns: the value returned by `operation`
+    /// - Throws: the error the `operation` has thrown (if any)
+    func withSpan<Return>(
```

Comment:
Should we make this `public`? If someone wanted to create a span inside a synchronous middleware it would be helpful.

Reply:
Err yeah they were all meant to be public, somehow missed that. Now fixed

Reply:
<img width="1840" alt="image" src="https://user-images.githubusercontent.com/16192401/214642397-1e271bc8-25d8-41dc-8848-10282ceb8b09.png">

Cool 👍 Here's how it would look like to have spans created in both sync and async middleware now.